lib/ansible/modules/network/ios/ios_acl.py

```diff
@@ -0,0 +1,946 @@
+#
+# vios#sh running-config | include access-list|ipv6 access-list|permit|deny
+# access-list 110 deny   tcp 198.51.100.0 0.0.0.255 eq telnet 198.51.110.0 0.0.0.255 eq www ack dscp ef ttl eq 10
+# access-list 150 deny   tcp 198.51.100.0 0.0.0.255 eq telnet 198.51.110.0 0.0.0.255 eq telnet syn dscp ef ttl eq 10
+
+# Using Deleted
+
+# Before state:
+# -------------
+#
+# vios#sh running-config | include access-list|ipv6 access-list|permit|deny
+# access-list 110 deny   tcp 198.51.100.0 0.0.0.255 eq telnet 198.51.110.0 0.0.0.255 eq www ack dscp ef ttl eq 10
+# access-list 150 deny   tcp 198.51.100.0 0.0.0.255 eq telnet 198.51.110.0 0.0.0.255 eq telnet syn dscp ef ttl eq 10
+
+- name: "Delete module attributes of given acl (Note: This won't delete the interface itself)"
+  ios_acl:
+    config:
+      - afi: ipv4
+        acls:
+          - name: 110
+    state: deleted
+
+# After state:
+# -------------
+#
+# vios#sh running-config | include access-list|ipv6 access-list|permit|deny
+# access-list 150 deny   tcp 198.51.100.0 0.0.0.255 eq telnet 198.51.110.0 0.0.0.255 eq telnet syn dscp ef ttl eq 10
+
+# Using Deleted without any config passed
+#"(NOTE: This will delete all of configured resource module attributes from each configured interface)"
+
+# Before state:
+# -------------
+#
+# vios#sh running-config | include access-list|ipv6 access-list|permit|deny
+# access-list 150 deny   tcp 198.51.100.0 0.0.0.255 eq telnet 198.51.110.0 0.0.0.255 eq telnet syn dscp ef ttl eq 10
+
+- name: "Delete module attributes of all acl (Note: This won't delete the interface itself)"
+  ios_acl:
+    state: deleted
+
+# After state:
+# -------------
+#
+# vios#sh running-config | include access-list|ipv6 access-list|permit|deny
+
+"""
+
+RETURN = """
+before:
+  description: The configuration as structured data prior to module invocation.
+  returned: always
+  type: list
+  sample: The configuration returned will always be in the same format of the parameters above.
+after:
+  description: The configuration as structured data after module completion.
+  returned: when changed
+  type: list
+  sample: The configuration returned will always be in the same format of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device
+  returned: always
+  type: list
+  sample: ['ip access-list extended 110', 'deny icmp 192.0.2.0 0.0.0.255 192.0.3.0 0.0.0.255 echo dscp ef ttl eq 10']
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.ios.argspec.acl.acl import AclArgs
+from ansible.module_utils.network.ios.config.acl.acl import Acl
+
+
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
+    required_if = [('state', 'merged', ('config',)),
+                   ('state', 'replaced', ('config',)),
+                   ('state', 'overridden', ('config',))]
+
+    module = AnsibleModule(argument_spec=AclArgs.argument_spec,
```

Comment:
`AclArgs.argument_spec` is returning `None`, which is causing `validate-modules` to stack trace. There's probably a bug in that class.

Reply:
@samdoran Thanks for reviewing and fixing the stack trace error, and in this case doc was returning None coz of the wrong indentation in the doc.
So, stack trace actually helped in triaging the issue and in the fix I can see you have returned empty dict if None which might skip the error scenarios, can we include any warnings for the users in this case when we encounter None.

Reply:
We could probably add another check for the argument spec being `None`.

Reply:
yes, that'd be a better fix for the same.